bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/connector/Executor.java

```diff
@@ -29,16 +30,26 @@
  * {@code Executor} Is the entry point from server connector side to ballerina side. After doing the dispatching and
  * finding the resource, server connector implementations can use this API to invoke Ballerina engine.
  *
- * @since 0.995
+ * @since 0.995.0
  */
 public class Executor {
 
+    /**
+     * This method will execute Ballerina resource in non-blocking manner. It will use Ballerina worker-pool for the
+     * execution and will return the connector thread immediately.
+     *
+     * @param service      to be executed.
+     * @param resourceName to be executed.
+     * @param callback     to be executed when execution completes.
+     * @param properties   to be passed to context.
+     * @param args         required for the resource.
+     */
     public static void submit(ObjectValue service, String resourceName, CallableUnitCallback callback,
-                              Map<String, Object> properties, Object... bValues) {
+                              Map<String, Object> properties, Object... args) {
         //TODO this is temp fix till we get the service.start() API
         Executors.newSingleThreadExecutor().submit(() -> {
             //TODO check the scheduler thread count
-            Object returnValues = service.call(new Strand(new Scheduler(4), properties), resourceName, bValues);
+            Object returnValues = service.call(new Strand(new Scheduler(4), properties), resourceName, args);
```

Comment:
Here we should use the existing scheduler to run the new strand.

Reply:
This is starting point of the ballerina thread execution. So we have to create new scheduler as don't have any existing ones